src/WebJobs.Script.WebHost/Diagnostics/DiagnosticEventLogger.cs

```diff
@@ -6,31 +6,47 @@
 using System.Linq;
 using Microsoft.Azure.WebJobs.Script.Config;
 using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
 
 namespace Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics
 {
-    public class DiagnosticEventLogger : ILogger
+    public class DiagnosticEventLogger : ILogger, IDisposable
     {
         private readonly IDiagnosticEventRepositoryFactory _diagnosticEventRepositoryFactory;
         private readonly IEnvironment _environment;
+        private readonly IDisposable _standbyChangeHandler;
+
         private IDiagnosticEventRepository _diagnosticEventRepository;
+        private bool _isEnabled;
 
-        public DiagnosticEventLogger(IDiagnosticEventRepositoryFactory diagnosticEventRepositoryFactory, IEnvironment environment)
+        public DiagnosticEventLogger(IDiagnosticEventRepositoryFactory diagnosticEventRepositoryFactory, IEnvironment environment,
+            IOptionsMonitor<StandbyOptions> standbyOptions)
         {
             _diagnosticEventRepositoryFactory = diagnosticEventRepositoryFactory;
             _environment = environment;
+
+            SetEnabledState(standbyOptions.CurrentValue);
+
+            // Re-evaluate whether this is enabled after specialization
+            if (standbyOptions.CurrentValue.InStandbyMode)
+            {
+                _standbyChangeHandler = standbyOptions.OnChange(o => SetEnabledState(o));
+            }
         }
 
-        public IDisposable BeginScope<TState>(TState state)
+        private void SetEnabledState(StandbyOptions options)
         {
-            return null;
+            _isEnabled = !options.InStandbyMode &&
+                !FeatureFlags.IsEnabled(ScriptConstants.FeatureFlagDisableDiagnosticEventLogging, _environment);
```

Comment:
Q: Do we need the feature flag check here? Looks like we do this check before registering this type to DI container (in `WebHostServiceCollectionExtension.AddWebJobsScriptHost`).

Reply:
it has to be checked here as well:
- Ideally, we'd never even register this, which is what the DI container checks for. This works in non-placeholder cases.
- However, in placeholder mode in the webhost, we have to always register it, then "disable" it after specialization and the env vars have changed. There's currently no way to adjust the registered WebHost services after we specialize... they need to all be registered at startup.